tools/diffscripts/diffapi.ps1

```diff
@@ -0,0 +1,203 @@
+	-Preview indicates you will compare the output to the last preview version
+
+.EXAMPLE
+.\diffapi
+
+Executes the commands assuming everything is cloned relative to the SDK repo and the AsmDiff tool is compiled using the arcade instructions.
+.EXAMPLE
+.\diffapi -AsmToolPath c:\tools\asmdifftool
+
+Specifies the location of the AsmDiff tool executable to run the diff commands.
+.EXAMPLE
+.\diffapi -SDKInternalsPath c:\repo\iot-sdks-internals
+
+Specifies the location of the SDK Internals repository if not cloned relative to the SDK repository.
+.EXAMPLE
+.\diffapi -Preview
+
+Executes the commands assuming everything is cloned relative to the SDK repo but will compare against the preview versions of the documents.
+.LINK
+https://github.com/azure/azure-iot-sdk-csharp
+
+#>
+
+Param(
+    [ValidateScript({
+            if( -Not ($_ | Test-Path -PathType Container) ){
+                throw "Folder $_ does not exist."
+            }
+            return $true
+        })]
+    # The path to the compiled AsmDiff tool found in the dotnet arcade (ex: c:\tools\asmdifftool)
+    [System.IO.FileInfo] $AsmToolPath = $null,
+    [ValidateScript({
+            if( -Not ($_ | Test-Path -PathType Container) ){
+                throw "Folder $_ does not exist."
+            }
+            return $true
+        })]
+    # The path of the iot-sdk-internals repository (ex: c:\repo\iot-sdks-internals)
+    [System.IO.FileInfo] $SDKInternalsPath = $null,
+    # Indicates you will compare the output to the last preview version instead of master
+    [switch] $Preview
+)
+
+
+$reporoot = (gi $pwd).Parent.Parent.FullName
+$baseroot = (gi $reporoot).Parent.FullName
+
+$asmtoolroot = (Join-Path -Path $baseroot -Child "\arcade\artifacts\bin\Microsoft.DotNet.AsmDiff\Debug\netcoreapp3.1\") 
+if ($AsmToolPath -ne $null) {
+    $asmtoolroot = $AsmToolPath
+}
+$asmtool = $asmtoolroot + "Microsoft.DotNet.AsmDiff.exe"
+
+
+$internalroot = Join-Path -Path $baseroot -Child "\iot-sdks-internals"
+
+if ($AsmToolPath -ne $null) {
+    $internalroot = $SDKInternalsPath
+}
+
+$compareDirectory = Join-Path -Path $internalroot -Child "\sdk_design_docs\CSharp\master"
+
+if ($Preview) {
+    $compareDirectory = Join-Path -Path $internalroot -Child "\sdk_design_docs\CSharp\preview"
+}
+
+# Hardcoded list of assembly names
+$filenames = @(
+    "Microsoft.Azure.Devices.Shared",
+    "Microsoft.Azure.Devices.Client",
+    "Microsoft.Azure.Devices",
+    "Microsoft.Azure.Devices.Provisioning.Client",
+    "Microsoft.Azure.Devices.Provisioning.Service",
+    "Microsoft.Azure.Devices.Provisioning.Transport.Amqp",
+    "Microsoft.Azure.Devices.Provisioning.Transport.Mqtt",
+    "Microsoft.Azure.Devices.Provisioning.Transport.Http",
+    "Microsoft.Azure.Devices.Provisioning.Security.Tpm"
+)
+
+# All of the files from the build
+$oldsetNames = @(
+    ((Join-Path -Path $reporoot -ChildPath "\shared\src\bin\Release\netstandard2.1\")+($filenames[0]+".dll")),
```

Comment:
Consider making the filename part of a `-AdditionalChildPath` parameter, instead of concatenating. Seems weird to use this utility but also strcat.

https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/join-path?view=powershell-7.1

Reply:
Looks like this is only available on PS 7.1 on my CORP machine it does not look like we install 7.1 by default. No need to take an upgrade just to get this to work.

```
❯ Get-Host


Name             : ConsoleHost
Version          : 5.1.19041.610
```